src/planner/update_plan.cpp

```diff
@@ -32,7 +39,7 @@ UpdatePlan::UpdatePlan(storage::DataTable *table,
   LOG_TRACE("Creating an Update Plan");
 }
 
-UpdatePlan::UpdatePlan(parser::UpdateStatement *parse_tree) {
+void UpdatePlan::Init(parser::UpdateStatement *parse_tree, std::vector<oid_t>& columns) {
```

Comment:
The name `Init` is ambiguous. This code is actually building the update plan, and the constructor code is initializing the scan plan. Something like `BuildUpdatePlan` sounds better?